webapp/tests/mocha/unit/translator.spec.js

```diff
@@ -0,0 +1,65 @@
+const BootstrapTranslator = require('../../../src/js/bootstrapper/translator'),
+      { expect } = require('chai');
+
+const translationData = {
+  en: {
+    LOAD_ASSETS:   'Loading assets!',
+    FETCH_INFO:    x => `Fetching info (${x} docs)!`,
+    EN_ONLY:       'Yay!',
+  },
+  es: {
+    LOAD_ASSETS:   'Cargando activos!',
+    FETCH_INFO:    x => `Descarga de datos (${x} documentos)!`,
+    ES_ONLY:      'Vamos!',
+  },
+};
+
+describe('Bootstrap Translator', () => {
+  it('Translate LOAD_ASSETS to English', () => {
+    const translator = new BootstrapTranslator('en', translationData);
+    expect(translator.translate('LOAD_ASSETS')).to.eq(translationData.en.LOAD_ASSETS);
+  });
+
+  it('Translate defaults and setLocale to English', () => {
+    const translator = new BootstrapTranslator(undefined, translationData);
+    expect(translator.translate('LOAD_ASSETS')).to.eq(translationData.en.LOAD_ASSETS);
+    translator.setLocale('es');
+    expect(translator.translate('LOAD_ASSETS')).to.eq(translationData.es.LOAD_ASSETS);
+  });
+
+  it('Translate to Spanish', () => {
+    const translator = new BootstrapTranslator('es', translationData);
+    expect(translator.translate('LOAD_ASSETS')).to.eq(translationData.es.LOAD_ASSETS);
+  });
+
+  it('Translate with embedded argument to Spanish', () => {
+    const translator = new BootstrapTranslator('es', translationData);
+    expect(translator.translate('FETCH_INFO', 35)).to.eq('Descarga de datos (35 documentos)!');
+  });
+
+  it('Missing translation falls back to English', () => {
+    const translator = new BootstrapTranslator('es', translationData);
+    expect(translator.translate('EN_ONLY')).to.eq(translationData.en.EN_ONLY);
```

Comment:
This appears to return the key instead of translating - shouldn't the expected translation be `'Yay!'`?

Reply:
Isn't the value of `translationData.en.EN_ONLY` equal to `Yay!`? 
Or are you saying the unit test should be written as the equivalent `expect(translator.translate('EN_ONLY')).to.eq('Yay!')`?

Reply:
Ah sorry, you're correct - I misread the code.